### PR TITLE
add immutability (if available) on ImageLess TextureRectangle

### PR DESCRIPTION
**Pull request by mp3butcher**

### CODE REVIEW

**Comment by mp3butcher** [comment]
comment removed

File: src/osg/TextureRectangle.cpp [hunk 1]
```diff
@@ -275,8 +275,14 @@ void TextureRectangle::apply(State& state) const
 
         applyTexParameters(GL_TEXTURE_RECTANGLE,state);
 
+        GLExtensions * extensions = state.get<GLExtensions>();
+        bool useTexStorage = extensions->isTextureStorageEnabled;
         // no image present, but dimensions at set so lets create the texture
-        glTexImage2D( GL_TEXTURE_RECTANGLE, 0, _internalFormat,
+        if(useTexStorage)
+            extensions->glTexStorage2D( GL_TEXTURE_RECTANGLE, 1, //TODO (_numMipmapLevels >0)?_numMipmapLevels:1,
```
[the remark this answers is not in the document]
Comment (mp3butcher) [reply src/osg/TextureRectangle.cpp]: sorry, I had forget this specificity of texrect

**Comment by openscenegraph** [comment]
I am going to hold back on merging this PR until we're resolved how to handle the internalFormat dependency issue is causing problems for others OSG users.

Perhaps only use glTextureStorage if internalFormat is set?  Or we have to provide an automatic mapping of sourceFormat to internalFormat.

**Comment by openscenegraph** [comment]
I an going to merge this PR, but I guess we may need to revist this code once we resolve how best to deal with sourceFormat mapping when and where it's needed.

**Comment by mp3butcher** [comment]
indeed

**Pull request merged**